From 122bdfcc5cd76d3ed24c2cb8791bea6e8e90d29e Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Thu, 8 Feb 2018 15:09:00 -0800
Subject: [PATCH 0259/1851] drm: xlnx: bridge: Fix the multiple definition
 error

commit 85c76cf798fa0226018c0a5e3bea1bab956241d5 from
https://github.com/Xilinx/linux-xlnx.git

The function definitions in header should be static inline when the config
is disabled, to avoid multiple definition error. While at it, fix
unmatched return type of xlnx_bridge_register().

Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_bridge.h | 26 ++++++++++++++------------
 1 file changed, 14 insertions(+), 12 deletions(-)

diff --git a/drivers/gpu/drm/xlnx/xlnx_bridge.h b/drivers/gpu/drm/xlnx/xlnx_bridge.h
index 3936c51cb9e4..8af32b53d23b 100644
--- a/drivers/gpu/drm/xlnx/xlnx_bridge.h
+++ b/drivers/gpu/drm/xlnx/xlnx_bridge.h
@@ -79,56 +79,58 @@ void xlnx_bridge_unregister(struct xlnx_bridge *bridge);
 
 struct xlnx_bridge_helper;
 
-int xlnx_bridge_helper_init(void)
+static inline inline int xlnx_bridge_helper_init(void)
 {
 	return 0;
 }
 
-void xlnx_bridge_helper_fini(void)
+static inline void xlnx_bridge_helper_fini(void)
 {
 }
 
-int xlnx_bridge_enable(struct xlnx_bridge *bridge)
+static inline int xlnx_bridge_enable(struct xlnx_bridge *bridge)
 {
 	if (bridge)
 		return -ENODEV;
 	return 0;
 }
 
-void xlnx_bridge_disable(struct xlnx_bridge *bridge)
+static inline void xlnx_bridge_disable(struct xlnx_bridge *bridge)
 {
 }
 
-int xlnx_bridge_set_input(struct xlnx_bridge *bridge,
-			  u32 width, u32 height, u32 bus_fmt)
+static inline int xlnx_bridge_set_input(struct xlnx_bridge *bridge,
+					u32 width, u32 height, u32 bus_fmt)
 {
 	if (bridge)
 		return -ENODEV;
 	return 0;
 }
 
-int xlnx_bridge_get_input_fmts(struct xlnx_bridge *bridge,
-			       const u32 **fmts, u32 *count)
+static inline int xlnx_bridge_get_input_fmts(struct xlnx_bridge *bridge,
+					     const u32 **fmts, u32 *count)
 {
 	if (bridge)
 		return -ENODEV;
 	return 0;
 }
 
-struct xlnx_bridge *of_xlnx_bridge_get(struct device_node *bridge_np)
+static inline struct xlnx_bridge *
+of_xlnx_bridge_get(struct device_node *bridge_np)
 {
 	return NULL;
 }
 
-void of_xlnx_bridge_put(struct xlnx_bridge *bridge)
+static inline void of_xlnx_bridge_put(struct xlnx_bridge *bridge)
 {
 }
 
-void xlnx_bridge_register(struct xlnx_bridge *bridge)
+static inline int xlnx_bridge_register(struct xlnx_bridge *bridge)
 {
+	return 0;
 }
 
-void xlnx_bridge_unregister(struct xlnx_bridge *bridge)
+static inline void xlnx_bridge_unregister(struct xlnx_bridge *bridge)
 {
 }
 
-- 
2.31.1

